daktari/checks/docker.py

```diff
@@ -0,0 +1,57 @@
+import logging
+import re
+from typing import Optional
+
+from daktari.check import Check, CheckResult
+from daktari.command_utils import get_stdout
+from daktari.os import OS
+
+
+class DockerInstalled(Check):
+    name = "docker.installed"
+
+    suggestions = {OS.GENERIC: "Install docker: https://docs.docker.com/get-docker/"}
+
+    def check(self) -> CheckResult:
+        return self.verify_install("docker")
+
+
+major_version_pattern = re.compile("([0-9]+)")
+
+
+def get_major_docker_version() -> Optional[int]:
+    raw_version = get_stdout("docker version --format '{{.Server.Version}}'")
```

Comment:
Could we do this via `docker --version` - on MacOS, `docker version` tries to talk to the Docker daemon, which isn't always running (not least because it's quite resource hungry)